Microsoft.DotNet.Interactive/KernelBase.cs

```diff
@@ -167,6 +168,11 @@ command switch
 
         public abstract bool TryGetVariable(string name, out object value);
 
+        public virtual Task<JObject> LspMethod(string methodName, JObject request)
```

Comment:
are we expecting the api to be opt in or otp out? @jonsequitur should this be abstract?

Reply:
Ultimately I don't care either way.  I made the method `virtual` simply so F#/pwsh don't have to implement this method simply to return `null` if they don't support LSP.

Reply:
`JObject` isn't an ideal return type. We should use strong types here and let serialization happen higher up.

Reply:
I just added a commit titled `strongly type LSP responses` that does just this; take a look.